src/main/java/abel/LinkedList.java

```diff
@@ -0,0 +1,128 @@
+package abel;
+
+/**
+ * Class that implements a simple Linked List.
+ *
+ * @param <T> Generic Object.
+ */
+public class LinkedList<T> implements List<T> {
+
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    /**
+     * Default Constructor.
+     */
+    public LinkedList() {
+        this.first = null;
+        this.last = null;
+        this.size = 0;
+    }
+
+    /**
+     * Method to add the first item.
+     *
+     * @param node Node object.
+     */
+    private void addFirst(final Node node) {
+        node.next = null;
+        this.first = node;
+        this.last = node;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void add(final T data) {
+        Node<T> node = new Node<>(data);
+        if (this.first == null) {
+            addFirst(node);
+        } else {
+            Node current = this.last;
+            this.last = node;
+            current.next = node;
+        }
+        this.size++;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public T get(int index) {
+        int count = 0;
+        Node current = first;
```

Comment:
```suggestion
        Node<T> current = first;
```
Use your T parameter so when you are doing this current.data you don't need to cast